Lib/tkinter/test/test_tkinter/test_misc.py

```diff
@@ -156,6 +156,33 @@ def callback():
         with self.assertRaises(tkinter.TclError):
             root.tk.call('after', 'info', idle1)
 
+    def test_winfo_rgb(self):
+        root = self.root
+        rgb = root.winfo_rgb
+
+        # Color name.
+        self.assertEqual(rgb('red'), (65535, 0, 0))
+        self.assertEqual(rgb('dark slate blue'), (18504, 15677, 35723))
+        # #RGB - extends each 4-bit hex value to be 16-bit.
+        self.assertEqual(rgb('#f00'),
+                         (int('ffff', 16), 0, 0))
+        # #RRGGBB - extends each 8-bit hex value to be 16-bit.
+        self.assertTrue(rgb('#483d8b') ==
+                        (int('4848', 16), int('3d3d', 16), int('8b8b', 16)) ==
+                        (18504, 15677, 35723))  # dark slate blue
+        # #RRRGGGBBB - uses 2 highest order bits for each color.
+        self.assertEqual(rgb('#123456789'),
+                         (int('1212', 16), int('4545', 16), int('7878', 16)))
+        # #RRRRGGGGBBBB - uses 2 highest order bits for each color.
+        self.assertEqual(rgb('#1234567800ff'),
+                         (int('1212', 16), int('5656', 16), int('0000', 16)))
+        # Invalid string.
+        with self.assertRaises(tkinter.TclError):
+            rgb('#1234567890')
+        # RGB triplet is invalid input.
+        with self.assertRaises(tkinter.TclError):
+            rgb((255, 0, 0))
+
```

Comment:
The main thrust of this issue is to update tkinter.colorchooser.Chooser._fixresult for 3.x.  So at least that method should be tested, with a test the fails currently. Since _fixresult does not access 'self', this can be done by extracting it from the class.  We do not even need to embed it in a mock class (which I have done for the non-GUI functions of some IDLE dialog classes).  But it would help to wrap it in a partial function.  Untested addition:
```
def test__fixresult(self):
    root = self.root
    _fixresult = tkinter.colorchooser.Chooser._fixresult
    def fix(result):
        return _fixresult(None, root, result)
    self.assertEqual(fix(''), (None, None))
    self.assertEqual(fix('#a0a0a0'), ((160, 160, 160), '#a0a0a0')
```
This completely covers _fixresult and with widget.winfo_rgb thoroughly tested, I think it is sufficient.  The second assertion fails now.

Reply:
I had actually written this initially, but because it was a new program, missed it when doing the commit.  :-(  Sorry about that.